### DISCUSSION TITLE
How do I know if status of the current suggestion ?

### DISCUSSION THREAD

**Discussion by hieulw** [monkoose/neocodeium#6] [Q&A] [answered]
How do I know if suggestion is show or not using neocodeium api ? I often map multiple functions into single keymap so I want an api to check condition. My config below is working but a bit hacky I think
```lua
      local codeium = require("neocodeium")

      vim.keymap.set("i", "<M-l>", function()
        if require("neocodeium.completer"):valid() then
          codeium.accept()
        else
          return "<Right>"
        end
      end, { expr = true, silent = true })
```

**Comment by monkoose** [comment]
Hello.
There are no exposed api for now. The best way to know if suggestion is really shown is by using `:h nvim_buf_get_extmarks()` with the plugin namespace (which can be get with `vim.api.nvim_get_namespaces().neocodeium_compl`. I would expose such api, but for now you can use something like this:
```lua
local codeium = require("neocodeium")

vim.keymap.set("i", "<M-l>", function()
  local ns = vim.api.nvim_get_namespaces().neocodeium_compl
  local extmarks = vim.api.nvim_buf_get_extmarks(0, ns, 0, -1, {})
  if #extmarks > 0 then
    codeium.accept()
  else
    return "<Right>"
  end
end, { expr = true, silent = true })
```

**Comment by monkoose** [comment]
On second thought I guess `completer:valid()` is fine and maybe better, number of suggestions label using the same namespace and it could show `0`.

**Comment by monkoose** [comment, marked as answer]
@hieulw I have added new api function `requre("neocodeium").visible()` you can use it instead of `completer:valid()`.
Internally it is just wrapper of `completer:valid()` (After some tests I have concluded that valid is perfectly fine for checking such status), but as part of public API is future proof for any internal changes.

**Reply by hieulw** [reply]
Great ! Thank you !
